vertica/datadog_checks/vertica/vertica.py

```diff
@@ -97,6 +97,9 @@ def check(self, instance):
 
             self._connection = connection
 
+        if self._connection.closed():
+            self._connection.reset_connection()
```

Comment:
Seems `reset_connection` will `close` and `startup_connection`.

https://github.com/vertica/vertica-python/blob/master/vertica_python/vertica/connection.py#L555-L557

Should we just `self.startup_connection()` ? Not sure if it an issue to call `self.close()` if the status is already closed.

Reply:
I think close can handle cleaning up additional stuff: https://github.com/vertica/vertica-python/blob/master/vertica_python/vertica/connection.py#L364-L366

Reply:
Yes, right, `close()` is going to reset all this.

https://github.com/vertica/vertica-python/blob/master/vertica_python/vertica/connection.py#L383-L391

```
    def reset_values(self):
        self.parameters = {}
        self.session_id = None
        self.backend_pid = None
        self.backend_key = None
        self.transaction_status = None
        self.socket = None
        self.address_list = _AddressList(self.options['host'], self.options['port'],
                                         self.options['backup_server_node'], self._logger)

```

So make sense to have call it too.